xgcm/test/test_padding.py

```diff
@@ -0,0 +1,1120 @@
+                0
+            ]  # ? not entirely sure why this is a list
+            expected = expected.pad({dim: widths}, "wrap")
+
+        # we intentionally strip all coords from padded arrays
+        expected = _strip_all_coords(expected)
+
+        result = pad(
+            data,
+            grid,
+            boundary="periodic",
+            boundary_width=boundary_width,
+        )
+        xr.testing.assert_allclose(expected, result)
+
+    def test_padding_mixed(self, boundary_width):
+        ds, coords, _ = datasets_grid_metric("C")
+        grid = Grid(ds, coords=coords)
+        data = ds.tracer
+
+        axis_padding_mapping = {"X": "periodic", "Y": "extend"}
+        method_mapping = {
+            "periodic": "wrap",
+            "extend": "edge",
+        }
+
+        # iterate over all axes
+        expected = data.copy(deep=True)
+        for ax, widths in boundary_width.items():
+            dim = grid._get_dims_from_axis(data, ax)[
+                0
+            ]  # ? not entirely sure why this is a list
+            expected = expected.pad(
+                {dim: widths}, method_mapping[axis_padding_mapping[ax]]
+            )
+
+        # we intentionally strip all coords from padded arrays
+        expected = _strip_all_coords(expected)
+
+        result = pad(
+            data,
+            grid,
+            boundary=axis_padding_mapping,
+            boundary_width=boundary_width,
+        )
+        xr.testing.assert_allclose(expected, result)
+
+
+# TODO: Add extrapolate, once we support it? Unless we do not support it anymore.
+
+
+class TestPaddingDefaults:
+    """Testing the behavior with default values for methods with additional input arguments"""
+
+    def test_padding_fill(self):
+
+        default_fill_value = 0
+        # TODO: Change this to a fixed value (nan?) in future releases.
+
+        ds, coords, _ = datasets_grid_metric("C")
+        grid = Grid(ds, coords=coords)
+        data = ds.tracer
+
+        expected = data.pad(
+            {"xt": (0, 1)}, "constant", constant_values=default_fill_value
+        )
+
+        # we intentionally strip all coords from padded arrays
+        expected = _strip_all_coords(expected)
+
+        result = pad(data, grid, boundary="fill", boundary_width={"X": (0, 1)})
+        xr.testing.assert_allclose(expected, result)
+
+    def test_padding_None(self):
+        "we currently expect the default padding to be periodic"
+        ds, coords, _ = datasets_grid_metric("C")
+        grid = Grid(ds, coords=coords)
+        data = ds.tracer
+
+        expected = data.pad({"xt": (0, 1)}, "wrap")
+
+        # we intentionally strip all coords from padded arrays
+        expected = _strip_all_coords(expected)
+
+        result = pad(data, grid, boundary=None, boundary_width={"X": (0, 1)})
+        xr.testing.assert_allclose(expected, result)
+
+
+# TODO: We should make both boundary and boundary width positional arguments. Then this class can be deleted.
```

Comment:
You mean make them positional arguments to the pad function? I think they should remain optional arguments to `apply_as_grid_ufunc`.